dockerfile-generator/dockerfile-gen-services/src/main/java/io/hyscale/dockerfile/gen/services/predicates/DockerfileGenPredicates.java

```diff
@@ -36,45 +36,40 @@
                 return false;
             }
             Dockerfile userDockerfile = null;
-            BuildSpec buildSpec = null;
             try {
                 userDockerfile = serviceSpec.get(
                         HyscaleSpecFields.getPath(HyscaleSpecFields.image, HyscaleSpecFields.dockerfile),
                         Dockerfile.class);
-                buildSpec = serviceSpec.get(
-                        HyscaleSpecFields.getPath(HyscaleSpecFields.image, HyscaleSpecFields.buildSpec),
-                        BuildSpec.class);
             } catch (HyscaleException e) {
                 logger.error("Error while fetching dockerfile from  image {}", e);
             }
 
-            if (userDockerfile == null && buildSpec == null) {
-                return true;
-            }
-
             if (userDockerfile != null) {
                 return true;
             }
 
-            if (stackAsServiceImage().test(buildSpec)) {
+            if (stackAsServiceImage().test(serviceSpec)) {
                 return true;
             }
 
             return false;
         };
     }
 
-    public static Predicate<Dockerfile> haveDockerfile() {
```

Comment:
why is this removed?

Reply:
It is not used anywhere, Also since its only a null check user can do that in place.